### ISSUE TITLE
Logger wip

### ISSUE THREAD

**Issue by mpclarkson** [qutheory/vapor#41]
I've written a basic logger which you can see [here] (https://github.com/mpclarkson/vapor/tree/logger) and would appreciate any input before taking it any further.

The relevant files are:

[LogDriver.swift](https://github.com/mpclarkson/vapor/blob/logger/Sources/LogDriver.swift)
This includes the `LoggerDriver` protocol and an example `ConsoleLoggerDriver` which simply prints logs to the console. 

The intention here is that you can implement any kind of logger (e.g. file, database etc) so long as it implements this protocol.

[Logger.swift](https://github.com/mpclarkson/vapor/blob/logger/Sources/Logger.swift)
This is a singleton... not sure what your position is on singletons but they are certainly widely used for loggers. Happy to change the implementation if you prefer otherwise. 

Currently the logger uses the `ConsoleLoggerDriver`, which is (current) injected in the private `init` for the example, but this would be better handled by registering a service provider (as discussed in #39 )

[Log.swift](https://github.com/mpclarkson/vapor/blob/logger/Sources/Log.swift)
This contains the `Level` enum, a `Loggable` protocol and a `Log` struct which implements this protocol. Currently, only date, level and message are required by the protocol but this could also include the file, function, line etc.

I have created an example log in `Request.init` as follows:

```swift
   //Testing logs
   let message = "Received \(method.rawValue) request for \(path)"
   let log = Log(level: .Info, message: message)
   Logger.sharedInstance.log(log)
```
Which prints the following to the console for every request:

`[2016-02-21 05:08:38 +0000] Info: Received GET request for /`

**Comment by tannernelson** [comment]
Great implementation so far.
I think the most Vapor way to do this would be an API like so.

```swift
Log.info("Informational log")
Log.error("something bad happened")
```

Where the `Log` class looks something like this

```swift
class Log {
  static var driver: LogDriver
  static func info(message: Any) { }
  ...etc
}
```

This way the logger implementation can be easily swapped out without affecting the interface for logging. 

Let me know what you think.

**Comment by mpclarkson** [comment]
Yep - makes sense. I will refactor it tomorrow my time.

**Comment by mpclarkson** [comment]
The logger now simply includes a `LoggerDriver` protocol, a default `ConsoleLogger` which implements this protocol and a `Log` class with static methods for each log level.

In the `Log` class you can:
- Override the default logger
- Define which levels to log by setting `[LogLevel]` to the static var `enabledLevels`.
- Log a `Custom` log level

For example:

```swift
Log.info("this is informational") //Only displays if enabledLevel contains .Info
Log.custom("this is a custom method", label: "MyLevel")

//Example output: [2016-02-21 23:14:36 +0000] INFO: Received GET request for /

```

```swift
public class Log {
    
    public static var driver: Logger = ConsoleLogger()
    public static var enabledLevels: [LogLevel] = LogLevel.all
    
    public static func verbose(message: String) {
        if Log.enabledLevels.contains(.Verbose) {
            driver.log(.Verbose, message: message)
        }
    }
  
 //other methods

```

Let me know...

**Comment by tannernelson** [comment]
This looks great. I'll test it out when I'm back on my computer.

**Comment by tannernelson** [comment]
Could you submit this code as a branch on `qutheory/vapor`?

**Comment by mpclarkson** [comment]
done

**Comment by mpclarkson** [comment]
Merged in #54
